pkg/codegen/nodejs/gen.go

```diff
@@ -1417,6 +1415,81 @@ func GeneratePackage(tool string, pkg *schema.Package, extraFiles map[string][]b
 		types.typeDetails, types.types = typeDetails, typeList
 	}
 
+	return modules, nil
+}
+
+// LanguageResource holds information about a resource to be used by downstream codegen.
+type LanguageResource struct {
+	*schema.Resource
+
+	Name       string             // The resource name (e.g., "FlowSchema")
+	Package    string             // The name of the package containing the resource definition (e.g., "flowcontrol.v1alpha1")
+	Properties []LanguageProperty // Properties of the resource
+}
+
+// LanguageProperty holds information about a resource property to be used by downstream codegen.
+type LanguageProperty struct {
+	ConstValue string // If set, the constant value of the property (e.g., "flowcontrol.apiserver.k8s.io/v1alpha1")
+	Name       string // The name of the property (e.g., "FlowSchemaSpec")
+	Package    string // The package path containing the property definition (e.g., "outputs.flowcontrol.v1alpha1")
+}
+
+// LanguageResources returns a map of resources that can be used by downstream codegen. The map
+// key is the resource schema token.
+func LanguageResources(tool string, pkg *schema.Package) (map[string]LanguageResource, error) {
+	resources := map[string]LanguageResource{}
+
+	if err := pkg.ImportLanguages(map[string]schema.Language{"nodejs": Importer}); err != nil {
+		return nil, err
+	}
+	info, _ := pkg.Language["nodejs"].(NodePackageInfo)
+
+	modules, err := generateModuleContextMap(tool, pkg, info)
+	if err != nil {
+		return nil, err
+	}
+
+	for modName, mod := range modules {
+		if modName == "" {
+			continue
+		}
```

Comment:
Why is this necessary? The root module may contain resources, functions, etc.

Reply:
I guess it's not required for k8s, but I'll handle this case downstream instead.